addons/payment/models/payment_provider.py

```diff
@@ -247,7 +247,10 @@ def _onchange_state_switch_is_published(self):
 
         :return: None
         """
-        self.is_published = self.state == 'enabled'
+        if self.state in ['enabled', 'test']:
+            self.is_published = True
+        elif self.state == 'disabled':
+            self.is_published = False
```

Comment:
This would change the current logic and make providers in test mode being published.